src/stores/tokens.ts

```diff
@@ -94,3 +109,57 @@ export function getToken(key: string) {
     const existing = get(tokens)
     return existing.find((t) => t.key === key)
 }
+
+export async function loadTokenMetadata(session: LinkSession) {
+    const records: Token[] = []
+
+    const sysToken = createTokenFromChainId(session.chainId, get(activePriceTicker))
+    records.push(sysToken)
+
+    AntelopeTokens.forEach((t) => {
+        const chain = chainConfig(session.chainId)
+
+        if (chain.id === t.chain) {
+            if (t.supply && t.supply.precision && t.symbol) {
+                const symbol: Asset.Symbol = Asset.Symbol.from(`${t.supply.precision},${t.symbol}`)
+                const token = {
+                    chainId: session.chainId,
+                    contract: t.account,
+                    symbol: symbol,
+                    name: t.metadata.name,
+                    logo: t.metadata.logo,
+                }
+
+                if (token.symbol.equals(sysToken.symbol) && token.contract === token.contract) {
+                    return
+                }
+
+                records.push({
+                    ...token,
+                    key: makeTokenKey(token),
+                })
+            }
+        }
+    })
+
+    tokens.set(records)
+}
+
+export async function loadTokenPrices(chain: ChainConfig, currentBalances: Balance[]) {
+    currentBalances.forEach((balance) => {
+        const token = getToken(balance.tokenKey)
+        if (token) {
+            const pairName = token.symbol.name.toLowerCase() + 'usd'
+            priceTicker(chain, pairName).value.subscribe((v) => {
```

Comment:
This leaks a subscription, any call to a stores subscribe must be matched with a unsubscribe

Reply:
I looked at this trying to figure out how to solve it, and was a bit confused. It's essentially the same as the `activePriceTicker` code, does that code need an unsubscribe as well?

Either way, I couldn't figure out how to use the ReadableStore that the price ticker returns in a loop like this. When would it unsubscribe from each of those? Would it use `get()` or something? What do you recommend here?

Reply:
If you subscribe without a matched unsubscribe call (i.e. never calling the function returned by subscribe) the store is never closed. If this happens elsewhere then that is a bug.

I would try to restructure the problem so that you could use a `derived` store here, failing that you need to return a function that unsubscribes to all price tickers when called and make sure that is called in the unsubscribe chain upstream.

Reply:
I gathered all the price tickers and unsubscribe them in the returned function.